tests/e2e/lib.sh

```diff
@@ -498,7 +517,6 @@ check_endpoint_availability() {
         sleep 1
     done
     die "Port ${target_port} did not become reachable in time"
-    exit 1
```

Comment:
Why remove this?

Reply:
My shellcheck was complaining since that line is unreachable. `die` already exits.